metal/ringbuf.h

```diff
@@ -0,0 +1,119 @@
+/* Copyright 2019 SiFive, Inc */
+/* SPDX-License-Identifier: Apache-2.0 */
+
+#ifndef METAL__RINGBUF_H
+#define METAL__RINGBUF_H
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+/*!
+ * @brief The handle for a ring buffer
+ */
+struct metal_ringbuf {
```

Comment:
We should also have another structure for item, that have a pointer and size of the item, which metal_ringbuf will include. Plus API to create this item, return item, its value, item size.

Reply:
That's a whole `sizeof(size_t)` per item additional overhead for a value which is constant for all items in the ring buffer. [FreeRTOS's Queues only require (capacity * item size) bytes of storage.](https://freertos.org/xQueueCreateStatic.html)